### PR TITLE
syslog: Handle ValueError raised while sending logs to syslog

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE

- Test Pull Request

### CODE REVIEW

File: lib/ansible/module_utils/basic.py [hunk 1]
```diff
@@ -1269,7 +1269,7 @@ def _log_to_syslog(self, msg):
                 facility = getattr(syslog, self._syslog_facility, syslog.LOG_USER)
                 syslog.openlog(str(module), 0, facility)
                 syslog.syslog(syslog.LOG_INFO, msg)
-            except TypeError as e:
+            except (ValueError, TypeError) as e:
```
Comment (mattclay) [inline lib/ansible/module_utils/basic.py]: What causes this? Is it a new behavior under Python 3.12?

Comment (jborean93) [reply lib/ansible/module_utils/basic.py]: It would be good to also update the changelog to include this info so users can know when this would be fixed.

Comment (Akasurde) [reply lib/ansible/module_utils/basic.py]: I am not sure if this is a new behavior or not, but checking on different versions- 

```
# python3
Python 3.11.5 (main, Aug 28 2023, 00:00:00) [GCC 13.2.1 20230728 (Red Hat 13.2.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import syslog
>>> syslog.syslog('\0')
```

```
# python3
Python 3.12.0 (main, Oct  2 2023, 00:00:00) [GCC 13.2.1 20230918 (Red Hat 13.2.1-3)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import syslog
>>> syslog.syslog('\0')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: embedded null character
```

Comment (jborean93) [reply lib/ansible/module_utils/basic.py]: Maybe just update the changelog saying handles `ValueError` for syslog introduced with python 3.12 or something.

**Review by jborean93 — APPROVED** [review]

**Comment by Akasurde** [comment]
For Posterity - 

```
08:17 The full traceback is:
08:17 Traceback (most recent call last):
08:17   File "<stdin>", line 133, in <module>
08:17   File "<stdin>", line 125, in _ansiballz_main
08:17   File "<stdin>", line 73, in invoke_module
08:17   File "<frozen runpy>", line 226, in run_module
08:17   File "<frozen runpy>", line 98, in _run_module_code
08:17   File "<frozen runpy>", line 88, in _run_code
08:17   File "/tmp/ansible_lineinfile_payload_c7rqa0dg/ansible_lineinfile_payload.zip/ansible/modules/lineinfile.py", line 637, in <module>
08:17   File "/tmp/ansible_lineinfile_payload_c7rqa0dg/ansible_lineinfile_payload.zip/ansible/modules/lineinfile.py", line 570, in main
08:17   File "/tmp/ansible_lineinfile_payload_c7rqa0dg/ansible_lineinfile_payload.zip/ansible/module_utils/basic.py", line 503, in __init__
08:17   File "/tmp/ansible_lineinfile_payload_c7rqa0dg/ansible_lineinfile_payload.zip/ansible/module_utils/basic.py", line 1377, in _log_invocation
08:17   File "/tmp/ansible_lineinfile_payload_c7rqa0dg/ansible_lineinfile_payload.zip/ansible/module_utils/basic.py", line 1344, in log
08:17   File "/tmp/ansible_lineinfile_payload_c7rqa0dg/ansible_lineinfile_payload.zip/ansible/module_utils/basic.py", line 1271, in _log_to_syslog
08:17 ValueError: embedded null character
08:17 fatal: [testhost]: FAILED! => {
08:17     "changed": false,
08:17     "module_stderr": "Traceback (most recent call last):\n  File \"<stdin>\", line 133, in <module>\n  File \"<stdin>\", line 125, in _ansiballz_main\n  File \"<stdin>\", line 73, in invoke_module\n  File \"<frozen runpy>\", line 226, in run_module\n  File \"<frozen runpy>\", line 98, in _run_module_code\n  File \"<frozen runpy>\", line 88, in _run_code\n  File \"/tmp/ansible_lineinfile_payload_c7rqa0dg/ansible_lineinfile_payload.zip/ansible/modules/lineinfile.py\", line 637, in <module>\n  File \"/tmp/ansible_lineinfile_payload_c7rqa0dg/ansible_lineinfile_payload.zip/ansible/modules/lineinfile.py\", line 570, in main\n  File \"/tmp/ansible_lineinfile_payload_c7rqa0dg/ansible_lineinfile_payload.zip/ansible/module_utils/basic.py\", line 503, in __init__\n  File \"/tmp/ansible_lineinfile_payload_c7rqa0dg/ansible_lineinfile_payload.zip/ansible/module_utils/basic.py\", line 1377, in _log_invocation\n  File \"/tmp/ansible_lineinfile_payload_c7rqa0dg/ansible_lineinfile_payload.zip/ansible/module_utils/basic.py\", line 1344, in log\n  File \"/tmp/ansible_lineinfile_payload_c7rqa0dg/ansible_lineinfile_payload.zip/ansible/module_utils/basic.py\", line 1271, in _log_to_syslog\nValueError: embedded null character\n",
08:17     "module_stdout": "",
08:17     "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
08:17     "rc": 1
08:17 }

```

File: changelogs/fragments/syslog_exception.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+---
+minor_changes:
```
Comment (jborean93) [inline changelogs/fragments/syslog_exception.yml]: Sorry shouldn't this be a bugfix not change, I assume we will be backporting to 2.16 for Python 3.12 support there?

**Review by jborean93 — APPROVED** [review]

**Comment by bcoca** [comment]
I would say we should avoid `embedded null character` .. but  .. lineinfile ...

**Pull request merged**